examples/chip-tool/commands/common/Commands.h

```diff
@@ -44,8 +44,10 @@ class Commands
     }
     int Run(int argc, char ** argv);
     int RunInteractive(const char * command, const chip::Optional<char *> & storageDirectory, bool advertiseOperational);
+    void SetCommandExitStatus(CHIP_ERROR status);
 
 private:
+    Command * mActiveCommand = nullptr;
```

Comment:
How does this handle interactive mode, where you have the overall "interactive start" command _and_ can have another command running under it?

Looks to me like this will cancel the "inner" command but then not actually quit interactive mode.  Is that the desired behavior?  If so, it might be best to document it.

Reply:
The current implementation is to exit the app directly. (No signal handler is hooked) But I think it seems more reasonable just to revoke the inner running command. For example, if the parameters are incorrect, user can just revoke this command and re-issue it again, instead of leaving the app and drop everything. User can have another ctrl+c to close the app.

Reply:
"another ctrl+c to close the app" won't work, since mActiveCommand will be wrong at that point, no?

Reply:
"another ctrl+c" will break the readline() in InteractiveStartCommand::GetCommand(). Even the signal handling function has been registered, it won't be called and the program will exit directly